src/test/java/com/hazelcast/aws/AwsDiscoveryStrategyTest.java

```diff
@@ -65,6 +66,18 @@ public void useCurrentRegion() {
         assertEquals("us-hz-1", awsConfig.getRegion());
     }
 
+    @Test
+    public void throwOnInvalidRegion() {
+        // given
+        AwsDiscoveryStrategy awsDiscoveryStrategy = spy(new AwsDiscoveryStrategy(Collections.emptyMap(), null, mockClient));
```

Comment:
As for my previous comment, how about making `validateRegion()` package-private and test it directly here? That would make this test way simpler. And you could test all the cases.